plutus-benchmark/values/bench/Bench.hs

```diff
@@ -0,0 +1,241 @@
+                , bench "deleteCoin - existing coin"
+                    $ nf (FlattenedValue.deleteCoin (encodeUtf8 "policy500") (encodeUtf8 "token999999")) fValue
+                , bench "byPolicyId - not found"
+                    $ nf (FlattenedValue.byPolicyId (encodeUtf8 "notHere")) fValue
+                , bench "byPolicyId - existing policy"
+                    $ nf (FlattenedValue.byPolicyId (encodeUtf8 "policy500")) fValue
+                , bench "byTokenName - not found"
+                    $ nf (FlattenedValue.byTokenName (encodeUtf8 "notHere")) fValue
+                , bench "byTokenName - existing token"
+                    $ nf (FlattenedValue.byTokenName (encodeUtf8 "token999999")) fValue
+                ]
+        , env setupEnvMM1 $ \ ~mmValue ->
+            bgroup "MMValue"
+                [ bench "insertCoin - new coin"
+                    $ nf (MMValue.insertCoin (encodeUtf8 "policy500") (encodeUtf8 "token10000") 200) mmValue
+                , bench "insertCoin - existing coin"
+                    $ nf (MMValue.insertCoin (encodeUtf8 "policy500") (encodeUtf8 "token999999") 200) mmValue
+                , bench "lookupCoin - not found"
+                    $ nf (MMValue.lookupCoin (encodeUtf8 "policy500") (encodeUtf8 "token10000")) mmValue
+                , bench "lookupCoin - existing coin"
+                    $ nf (MMValue.lookupCoin (encodeUtf8 "policy500") (encodeUtf8 "token999999")) mmValue
+                , bench "deleteCoin - not found"
+                    $ nf (MMValue.deleteCoin (encodeUtf8 "policy500") (encodeUtf8 "token10000")) mmValue
+                , bench "deleteCoin - existing coin"
+                    $ nf (MMValue.deleteCoin (encodeUtf8 "policy500") (encodeUtf8 "token999999")) mmValue
+                , bench "byPolicyId - not found"
+                    $ nf (MMValue.byPolicyId (encodeUtf8 "notHere")) mmValue
+                , bench "byPolicyId - existing policy"
+                    $ nf (MMValue.byPolicyId (encodeUtf8 "policy500")) mmValue
+                , bench "byTokenName - not found"
+                    $ nf (MMValue.byTokenName (encodeUtf8 "notHere")) mmValue
+                , bench "byTokenName - existing token"
+                    $ nf (MMValue.byTokenName (encodeUtf8 "token999999")) mmValue
+                ]
+        , env setupEnvNested2 $ \ ~vals ->
+            bgroup "NestedValue"
+                [ bench "union - two different values"
+                    $ nf (uncurry NestedValue.union) vals
+                , bench "valueContains - is not sub-value"
+                    $ nf (uncurry NestedValue.valueContains) vals
+                ]
+        , env setupEnvFlattened2 $ \ ~vals ->
+            bgroup "FlattenedValue"
+                [ bench "union - two different values"
+                    $ nf (uncurry FlattenedValue.union) vals
+                , bench "valueContains - is not sub-value"
```

Comment:
I just realised `contains` is broken for all these implementations, redoing this.

Reply:
Fixed ^ and realized I forgot to push latest changes! This is updated now, I should take another look and regenerate the benchmark report